Modules/_abc.c

```diff
@@ -0,0 +1,639 @@
+       Stage 1: direct abstract methods.
+       (It is safe to assume everything is fine since type.__new__ succeeded.) */
+    ns = PyTuple_GET_ITEM(args, 2);
+    items = PyMapping_Items(ns); /* TODO: Fast path for exact dicts with PyDict_Next */
+    for (pos = 0; pos < PySequence_Size(items); pos++) {
+        item = PySequence_GetItem(items, pos);
+        key = PyTuple_GetItem(item, 0);
+        value = PyTuple_GetItem(item, 1);
+        is_abstract = _PyObject_GetAttrId(value, &PyId___isabstractmethod__);
+        if (!is_abstract) {
+            if (!PyErr_ExceptionMatches(PyExc_AttributeError)) {
+                Py_DECREF(item);
+                Py_DECREF(items);
+                goto error;
+            }
+            PyErr_Clear();
+            Py_DECREF(item);
+            continue;
+        }
+        if (is_abstract == Py_True && PySet_Add(abstracts, key) < 0) {
+            Py_DECREF(item);
+            Py_DECREF(items);
+            goto error;
+        }
+        Py_DECREF(item);
+    }
+    Py_DECREF(items);
+    /* Stage 2: inherited abstract methods. */
+    bases = PyTuple_GET_ITEM(args, 1);
+    for (pos = 0; pos < PyTuple_Size(bases); pos++) {
+        item = PyTuple_GET_ITEM(bases, pos);
+        base_abstracts = _PyObject_GetAttrId(item, &PyId___abstractmethods__);
+        if (!base_abstracts) {
+            if (!PyErr_ExceptionMatches(PyExc_AttributeError)) {
+                goto error;
+            }
+            PyErr_Clear();
+            continue;
+        }
+        if (!(iter = PyObject_GetIter(base_abstracts))) {
+            goto error;
+        }
+        while ((key = PyIter_Next(iter))) {
+            value = PyObject_GetAttr((PyObject *)result, key);
+            if (!value) {
+                if (!PyErr_ExceptionMatches(PyExc_AttributeError)) {
+                    Py_DECREF(key);
+                    Py_DECREF(iter);
+                    goto error;
+                }
+                PyErr_Clear();
+                Py_DECREF(key);
+                continue;
+            }
+            is_abstract = _PyObject_GetAttrId(value, &PyId___isabstractmethod__);
+            if (!is_abstract) {
+                if (!PyErr_ExceptionMatches(PyExc_AttributeError)) {
+                    Py_DECREF(key);
+                    Py_DECREF(value);
+                    Py_DECREF(iter);
+                    goto error;
+                }
+                PyErr_Clear();
+                Py_DECREF(key);
+                Py_DECREF(value);
+                continue;
+            }
+            if (is_abstract == Py_True && PySet_Add(abstracts, key) < 0) {
+                Py_DECREF(key);
+                Py_DECREF(value);
+                Py_DECREF(iter);
+                goto error;
+            }
+            Py_DECREF(key);
+            Py_DECREF(value);
+        }
+        Py_DECREF(iter);
+    }
+    if (_PyObject_SetAttrId((PyObject *)result, &PyId___abstractmethods__, abstracts) < 0) {
```

Comment:
Doesn't this cause a change in behavior; __abstractmethods__ used to be a `frozenset`, but would now be a regular mutable set